recipe/0001-patch-setup-conda-fftw-for-cross.patch

```diff
@@ -0,0 +1,106 @@
+-        if path in os.environ:
+-            for dir in os.environ[path].split(':'):
+-                try_libdirs.append(dir)
+-
+-    # The user's home directory is often a good place to check.
+-    try_libdirs.append(os.path.join(os.path.expanduser("~"),"lib"))
+-
+-    # If the above don't work, the fftw3 module may have the right directory.
+-    try:
+-        import fftw3
+-        try_libdirs.append(fftw3.lib.libdir)
+-    except ImportError:
+-        pass
+-
+     if sys.platform == "darwin":
+         lib_ext = '.dylib'
+     else:
+         lib_ext = '.so'
+     name = 'libfftw3' + lib_ext
+-    if output: print("Looking for ",name)
+-    tried_dirs = set()  # Keep track, so we don't try the same thing twice.
+-    for dir in try_libdirs:
+-        if dir == '': continue  # This messes things up if it's in there.
+-        if dir in tried_dirs: continue
+-        else: tried_dirs.add(dir)
+-        if not os.path.isdir(dir): continue
+-        libpath = os.path.join(dir, name)
+-        if not os.path.isfile(libpath): continue
+-        if output: print("  ", dir, end='')
+-        try:
+-            lib = ctypes.cdll.LoadLibrary(libpath)
+-            if output: print("  (yes)")
+-            return libpath
+-        except OSError as e:
+-            if output: print("  (no)")
+-            # Some places use lib64 rather than/in addition to lib.  Try that as well.
+-            if dir.endswith('lib') and os.path.isdir(dir + '64'):
+-                dir += '64'
+-                try:
+-                    libpath = os.path.join(dir, name)
+-                    if not os.path.isfile(libpath): continue
+-                    lib = ctypes.cdll.LoadLibrary(libpath)
+-                    if output: print("  ", dir, "  (yes)")
+-                    return libpath
+-                except OSError:
+-                    pass
+-    try:
+-        libpath = ctypes.util.find_library('fftw3')
+-        if libpath == None:
+-            raise OSError
+-        if os.path.split(libpath)[0] == '':
+-            # If the above doesn't return a real path, try this instead.
+-            libpath = ctypes.util._findLib_gcc('fftw3')
+-            if libpath == None:
+-                raise OSError
+-        libpath = os.path.realpath(libpath)
+-        lib = ctypes.cdll.LoadLibrary(libpath)
+-    except Exception as e:
+-        print("Could not find fftw3 library.  Make sure it is installed either in a standard ")
+-        print("location such as /usr/local/lib, or the installation directory is either in ")
+-        print("your LIBRARY_PATH or FFTW_DIR environment variable.")
+-        raise
+-    else:
+-        dir, name = os.path.split(libpath)
+-        if output:
+-            if dir == '': dir = '[none]'
+-            print("  ", dir, "  (yes)")
+-        return libpath
++    
++    libpath = os.path.realpath(os.path.join(sys.prefix, 'lib', name))
+ 
++    return libpath
```

Comment:
I guess I should add an option upstream to force a specific FFTW_DIR that would just trust the environment variable and not do these checks.  Maybe FORCE_FFTW_DIR?

Then for cross compiling, you could just do
```
FORCE_FFTW_DIR=$PREFIX python setup.py build_shared_clib
```
Then we wouldn't run the risk of this patch becoming out of date down the line if I change anything herein.

Reply:
If you'd like go for it.

Reply:
https://github.com/GalSim-developers/GalSim/commit/92b936a02058518ce40d89ebbd529321a74bbebb

On the next release version, we can remove this patch, and instead run:
```
FFTW_DIR=$PREFIX/lib python setup.py build_shared_clib
```
When FFTW_DIR is given, it will trust it even if the library cannot be loaded.